src/components/Calculator.js

```diff
@@ -1,22 +1,227 @@
-import Buttons from './button';
-import buttondata from './buttondata';
-
-const Calculator = () => {
-  const Button = buttondata
-    .map((btn) => (
-      <Buttons
-        key={btn.id}
-        val={btn.value}
-      />
-    ));
-  return (
-    <div className="calc">
-      <div className="display"><span>0</span></div>
-      <div className="btns">
-        {Button}
+import React from 'react';
+import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.Compute = this.Compute.bind(this);
+    this.updateScreen = this.updateScreen.bind(this);
+  }
+
```

Comment:
- [ ] As it is a [requirment ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md)for this project, "`Practice what you have learned until now: use useState and props`", please ensure the Calculator app handles state with `useState`, something like so: 

```jsx
import React, { useState } from 'react';
import calculate from ../logic; // Import math logic

function Calculator() {
    const [total, setTotal] = useState(0);
    return {
      <div>
       // Calculator app 
      </div>
    };
}
```

Please kindly fix that.